integration-tests/opentelemetry-logging/src/test/java/io/quarkus/logging/opentelemetry/it/LoggingResourceTest.java

```diff
@@ -0,0 +1,40 @@
+package io.quarkus.logging.opentelemetry.it;
+
+import static io.restassured.RestAssured.given;
+import static org.awaitility.Awaitility.await;
+import static org.hamcrest.Matchers.is;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.time.Duration;
+import java.util.List;
+
+import jakarta.inject.Inject;
+
+import org.junit.jupiter.api.Test;
+
+import io.opentelemetry.sdk.logs.data.LogRecordData;
+import io.opentelemetry.sdk.testing.exporter.InMemoryLogRecordExporter;
+import io.quarkus.test.junit.QuarkusTest;
+
+@QuarkusTest
+public class LoggingResourceTest {
+    @Inject
+    InMemoryLogRecordExporter exporter;
+
+    @Test
+    public void testHelloEndpoint() {
+        // This will create 1 log, quarkus startup already created 3 logs so we should have 4 of them.
+        given()
+                .when().get("/logging-opentelemetry")
+                .then()
+                .statusCode(200)
+                .body(is("Hello logging-opentelemetry"));
+
+        // Wait for logs to be available as everything is async
+        await().atMost(Duration.ofSeconds(10)).until(() -> exporter.getFinishedLogRecordItems().size() == 4);
+
+        List<LogRecordData> items = exporter.getFinishedLogRecordItems();
+        assertEquals(4, items.size());
+        assertEquals("Hello World", items.get(3).getBody().asString());
```

Comment:
We need to test many other combinations like exceptions, stack traces, log levels ...
Many of those tests can be implemented in the deployment project.